cmd/controller/main.go

```diff
@@ -284,6 +286,7 @@ func parseEnvFlags() config {
 	pflag.String(sidecarCPURequestFlag, viper.GetString(sidecarCPURequestFlag), "Flag to overwrite the GameServer sidecar container's cpu request. Can also use SIDECAR_CPU_REQUEST env variable")
 	pflag.String(sidecarMemoryLimitFlag, viper.GetString(sidecarMemoryLimitFlag), "Flag to overwrite the GameServer sidecar container's memory limit. Can also use SIDECAR_MEMORY_LIMIT env variable")
 	pflag.String(sidecarMemoryRequestFlag, viper.GetString(sidecarMemoryRequestFlag), "Flag to overwrite the GameServer sidecar container's memory request. Can also use SIDECAR_MEMORY_REQUEST env variable")
+	pflag.Int32(sidecarRunAsUserFlag, 1000, "Flag to indicate the GameServer sidecar container's UID. Can also use SIDECAR_RUN_AS_USER env variable")
```

Comment:
Actually I don't really understand how this line works, wherever a flag is an int, the existing code always puts an integer as the second argument instead of using viper.GetInt32(). I thought you need to call viper in order to retrieve the value in the template controller.yaml file?

Reply:
The `controller.yaml` file is specifying how the container is built. In our case it runs a container with various environment variables, including `SIDECAR_RUN_AS_USER`, which gets bound below (line 321). We actually never use the flags, just the env variables - but we have the flags here because people do deploy Agones other ways.